File: backend/src/main/java/com/andrew/hdss/models/enums/LocationType.java
package com.andrew.hdss.models.enums;

public enum LocationType {
    COUNTRY(0),
    COUNTY(1),
    SUB_COUNTY(2),
    LOCATION(3),
    SUB_LOCATION(4);

    private final int level;

    LocationType(int level) {
        this.level = level;
    }

    public int getLevel() {
        return level;
    }

    public LocationType parentType() {
        if (level == 0) return null;
        return values()[level - 1];
    }
}

File: android/app/src/main/java/com/andrew/hdss/data/models/enums/LocationType.kt
package com.andrew.hdss.data.models.enums

enum class LocationType(val level:Int) {
    COUNTRY(0),
    COUNTY(1),
    SUB_COUNTY(2),
    LOCATION(3),
    SUB_LOCATION(4);

    fun parentType(): LocationType? =
        if (level == 0) null else entries.first { it.level == level - 1 }
}
